checkov/common/images/image_referencer_manager.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+
+from checkov.common.images.image_referencer_provider import Provider
+
+from typing import Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from checkov.common.images.image_referencer import Image
+
+
+class ImageReferencerManager:
+
```

Comment:
you can define here `__slots__` so you don't need to do it in each of the subclasses.